### PR TITLE
Added timing options

### PR DESCRIPTION
**Pull request by bee-san**
Times how long it takes RustScan to run

### CODE REVIEW

File: src/scanner/mod.rs [hunk 1]
```diff
@@ -21,12 +21,13 @@ use std::{
 /// Timeout is the time RustScan should wait before declaring a port closed. As datatype Duration.
 /// Quiet is whether or not RustScan should print things, or wait until the end to print only open ports.
 #[cfg(not(tarpaulin_include))]
+#[derive(Debug)]
 pub struct Scanner {
-    ips: Vec<IpAddr>,
+    pub ips: Vec<IpAddr>,
```
Comment (bee-san) [inline src/scanner/mod.rs:26]: Why are these made public? :)

File: src/scanner/mod.rs [hunk 2]
```diff
@@ -105,16 +116,21 @@ impl Scanner {
...
-                            socket.ip().to_string(),
-                            "Host unreachable! Make sure the ip you target is online and you have a route to reach it. Exiting."
-                                .to_string()
-                                .red()
-                        );
-                        std::process::exit(1);
+                    if e.to_string().contains("No route to host")
+                        || e.to_string().contains("Network is unreachable")
+                    {
+                        // println!(
```
Comment (bee-san) [inline src/scanner/mod.rs:122]: delete unused code!

File: src/scanner/mod.rs [hunk 3]
```diff
@@ -140,6 +156,8 @@ impl Scanner {
             async move { TcpStream::connect(socket).await },
         )
         .await?;
+        // let newtimeout = &self.timeout;
```
Comment (bee-san) [inline src/scanner/mod.rs:159]: delete unused code :D

**Review by bee-san — COMMENTED** [review]
Nice! Love the statistics. Also, can you squash all of your commits to 1 commit? There's a link in the wiki to how to do this :)

https://github.com/RustScan/RustScan/wiki/Contributing-Tips#git-tips

File: src/scanner/mod.rs (hunk 1 above)
Comment (bergabman) [reply src/scanner/mod.rs:26]: I made them public so we can calculate average socket connection time based on number of ips, ports and runtime.

File: src/scanner/mod.rs (hunk 2 above)
Comment (bergabman) [reply src/scanner/mod.rs:122]: Deleted.

File: src/scanner/mod.rs (hunk 3 above)
Comment (bergabman) [reply src/scanner/mod.rs:159]: Deleted.

**Pull request closed without merging**